pkg/awsiamauth/awsiamauth.go

```diff
@@ -92,12 +92,13 @@ func (a *AwsIamAuth) GenerateManifestForUpgrade(clusterSpec *cluster.Spec) ([]by
 	return a.templateBuilder.GenerateManifest(clusterSpec, uuid.Nil)
 }
 
-func (a *AwsIamAuth) GenerateCertKeyPairSecret() ([]byte, error) {
+func (a *AwsIamAuth) GenerateCertKeyPairSecret(bootstrapClusterName string) ([]byte, error) {
```

Comment:
call this 'clusterName' because it could be either bootstrap or workload cluster?

Reply:
It should always be the bootstrap cluster (so KinD or management). I can add a comment 👍

Reply:
non-blocking but I think [bootstrap cluster](https://cluster-api.sigs.k8s.io/reference/glossary.html#bootstrap-cluster) refers to a temporary cluster that is used to provision a long lived cluster. In our case, this would be the kind cluster. The permanent management cluster is never a bootstrap cluster.
`managementClusterName` would be more appropriate since the bootstrap cluster is also technically a management cluster for most of its duration.
I don't really have a strong opinion on this, so I will leave it up to you guys to decide.